test/Microsoft.AspNetCore.Server.KestrelTests/MemoryPoolIteratorTests.cs

```diff
@@ -999,6 +966,34 @@ public void TestSeekIteratorLimitAcrossBlocks(string input, char seek, char limi
             }
         }
 
+        [Fact]
+        public void EmptyIteratorBehaviourIsValid()
+        {
+            const byte byteCr = (byte) '\n';
+            ulong longValue;
+            var end = default(MemoryPoolIterator);
+
+            Assert.False(default(MemoryPoolIterator).TryPeekLong(out longValue));
+            Assert.False(default(MemoryPoolIterator).Put(byteCr));
+            Assert.Null(default(MemoryPoolIterator).GetAsciiString(ref end));
+            Assert.Null(default(MemoryPoolIterator).GetUtf8String(ref end));
+            // Assert.Equal doesn't work for default(ArraySegments)
+            Assert.True(default(MemoryPoolIterator).GetArraySegment(end).Equals(default(ArraySegment<byte>)));
+            Assert.True(default(MemoryPoolIterator).IsDefault);
+            Assert.True(default(MemoryPoolIterator).IsEnd);
+            Assert.Equal(default(MemoryPoolIterator).Take(), -1);
+            Assert.Equal(default(MemoryPoolIterator).Peek(), -1);
+            Assert.Equal(default(MemoryPoolIterator).Seek(byteCr), -1);
+            Assert.Equal(default(MemoryPoolIterator).Seek(byteCr, ref end), -1);
+            Assert.Equal(default(MemoryPoolIterator).Seek(byteCr, byteCr), -1);
+            Assert.Equal(default(MemoryPoolIterator).Seek(byteCr, byteCr, byteCr), -1);
+
+            default(MemoryPoolIterator).CopyFrom(default(ArraySegment<byte>));
```

Comment:
No. SocketOutput.ProducingStart() returns default(MemoryPoolIterator) after the socket is closed. Frame.CreateResponseHeader() is not hardened against CopyFrom() throwing. Much like SocketOutput.WriteAsync() without a CancellationToken(), CopyFrom() on the return value of ProducingStart()  should always succeed.